ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java

```diff
@@ -539,6 +552,9 @@ public void removeFromGroup(@Nonnull final User user, @Nonnull final Group group
     private int enqueueGroupsByFilter(@Nonnull final Collection<Group> groups, @Nullable BaseFilter filter,
                                       @Nullable final Dn baseSearchDn, final int limit,
                                       final boolean restrictMaxAllowedResults) throws DaoException {
+        assertValidBaseGroupDn();
+        assert baseGroupDn != null;
+
         // require provided base dn be descendant of (or identical to) groups base dn, under threat of exception
         final Dn searchDn = MoreObjects.firstNonNull(baseSearchDn, baseGroupDn);
         if (!searchDn.isDescendantOfOrEqualTo(baseGroupDn)) {
```

Comment:
`assertValidGroupDn(baseGroupDn)` instead of:

```
        if (!searchDn.isDescendantOfOrEqualTo(baseGroupDn)) {
            throw new IllegalArgumentException(baseSearchDn + " must be descendant of (or identical to) " +
                    baseGroupDn);
        }
```